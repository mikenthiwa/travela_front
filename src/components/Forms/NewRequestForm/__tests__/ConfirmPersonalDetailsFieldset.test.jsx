import React from 'react';
import { shallow, mount } from 'enzyme';
import ConfirmPersonalDetailsFieldset from '../FormFieldsets/ConfirmPersonalDetails';

describe('<ConfirmPersonalDetailsFieldset/>',()=> {
  const props = {
    managers:['Mananger1'],
    occupations:['Software Developer'],
    departments: [{id:'dept', text:'finance', parentId: null}],
    collapsible: jest.fn(),
    collapse: false,
    userDetails: jest.fn(),
    nextStep: jest.fn(),
    values: {},
    savePersonalDetails: jest.fn(),
    onChangeAutoSuggestion: jest.fn(),
    hasBlankFields: true,
    getUserDetails: jest.fn(),
  };
  it('renders correctly', () => {
    const wrapper = shallow(<ConfirmPersonalDetailsFieldset {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('tests handleDisableInputs', ()=>{
    const wrapper = shallow(<ConfirmPersonalDetailsFieldset {...props} />);
    const instance = wrapper.instance();
    instance.handleDisableInputs('clicked');
    expect(wrapper.state('disableInputs')).toBe('disable-details');
  });
  it('tests onChangeAutoSuggestion inputField for role', () => {
    const event = { target: { value: 'yes' } };
    const wrapper = shallow(<ConfirmPersonalDetailsFieldset {...props} />);
    wrapper.find('Input#your-role').simulate('change', event);
    expect(wrapper.instance().props.onChangeAutoSuggestion).toHaveBeenCalledTimes(1);

  });
  it('tests onChangeAutoSuggestion inputField for manager', () => {
    const event = { target: { value: 'yes' } };
    const wrapper = shallow(<ConfirmPersonalDetailsFieldset {...props} />);
    wrapper.find('Input#your-manager').simulate('change', event);
    expect(wrapper.instance().props.onChangeAutoSuggestion).toHaveBeenCalledTimes(2);

  });

  it('tests onChangeAutoSuggestion inputField for Location', () => {
    const event = { target: { value: 'yes' } };
    const wrapper = shallow(<ConfirmPersonalDetailsFieldset {...props} />);
    wrapper.find('Input#user-location').simulate('change', event);
    expect(wrapper.instance().props.onChangeAutoSuggestion).toHaveBeenCalledTimes(3);

  });
  it('tests onChangeAutoSuggestion inputField for department', () => {
    const event = { target: { value: 'yes' } };
    const wrapper = shallow(<ConfirmPersonalDetailsFieldset {...props} />);
    wrapper.find('Input#your-department').simulate('change', event);
    expect(wrapper.instance().props.onChangeAutoSuggestion).toHaveBeenCalledTimes(3);

  });
  it('invalidates wrong input', () => {
    const invalidProps = {
      ...props,
      title: '',
    };
    const wrapper = shallow(<ConfirmPersonalDetailsFieldset {...invalidProps} />);
    const result = wrapper.find('#submit');
    expect(result.props().disabled).toEqual(true);
  });
  it('validtes correct input', () => {
    const wrapper = shallow(<ConfirmPersonalDetailsFieldset {...props} />);
    const result = wrapper.find('#submit');
    expect(result.props().disabled).toEqual(true);
  });
  it('It makes a call when the next button is click', () => {
    const event = { target: { value: 'yes' } };
    const wrapper = shallow(<ConfirmPersonalDetailsFieldset {...props} />);
    wrapper.find('button').simulate('click', event);
    expect(props.nextStep).toHaveBeenCalledTimes(1);
  });
});
