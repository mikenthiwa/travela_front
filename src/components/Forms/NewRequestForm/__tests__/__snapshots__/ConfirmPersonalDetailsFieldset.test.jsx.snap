// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ConfirmPersonalDetailsFieldset/> renders correctly 1`] = `
<fieldset
  className="personal-details null"
>
  <div
    className=""
  >
    <div
      className="personal-rectangle-details-items"
    >
      <div
        className="input-group-details mdl-grid "
      >
        <div
          className="spaces-right"
        >
          <Input
            autoComplete="off"
            className="request_dropdown"
            disabled={false}
            label="Your Name"
            labelNote=""
            name="name"
            onChange={null}
            required={false}
            type="filter-dropdown-select"
          />
        </div>
        <div
          className="spaces-right"
        >
          <Input
            autoComplete="off"
            choices={
              Array [
                "Female",
                "Male",
              ]
            }
            className=""
            disabled={false}
            label="What is your gender?"
            labelNote=""
            name="gender"
            onChange={null}
            required={false}
            type="onboarding-button-toggler"
          />
        </div>
      </div>
      <div
        className="input-group-details mdl-grid"
      >
        <div
          className="spaces-right"
        >
          <Input
            autoComplete="off"
            choices={Array []}
            className="request_dropdown your-department"
            disabled={false}
            id="your-department"
            label="What is your department?"
            labelNote=""
            name="department"
            onChange={[Function]}
            required={false}
            size=""
            type="filter-dropdown-select"
          />
        </div>
        <div
          className="spaces-right"
        >
          <Input
            autoComplete="off"
            choices={
              Array [
                undefined,
              ]
            }
            className="request_dropdown  your-role"
            id="your-role"
            label="What is your role?"
            labelNote=""
            name="role"
            onChange={[Function]}
            required={false}
            size=""
            type="filter-dropdown-select"
          />
        </div>
      </div>
      <div
        className="input-group-details mdl-grid"
      >
        <div
          className="spaces-right mdl-cell mdl-cell--4-col mdl-cell--6-col-tablet mdl-cell--12-col-phone"
        >
          <Input
            autoComplete="off"
            choices={
              Array [
                undefined,
              ]
            }
            className="request_dropdown your-manager"
            id="your-manager"
            label="Who is your Line Manager?"
            labelNote=""
            name="manager"
            onChange={[Function]}
            required={false}
            size=""
            type="filter-dropdown-select"
          />
        </div>
        <div
          className="spaces-right mdl-cell mdl-cell--4-col mdl-cell--6-col-tablet mdl-cell--12-col-phone"
        >
          <Input
            autoComplete="off"
            choices={Array []}
            className="request_dropdown user-location"
            disabled={false}
            id="user-location"
            label="Where is your location?"
            labelNote=""
            name="location"
            onChange={[Function]}
            required={false}
            size=""
            type="filter-dropdown-select"
          />
        </div>
      </div>
    </div>
    <div
      className="request-submit-area-details"
    >
      <button
        className="bg-btn bg-btn--active-1"
        disabled={true}
        id="submit"
        onClick={[Function]}
        type="submit"
      >
        <ButtonLoadingIcon
          buttonText=""
          isLoading={false}
        />
      </button>
    </div>
  </div>
</fieldset>
`;
